relay-server/src/actors/processor.rs

```diff
@@ -2715,8 +2732,9 @@ mod tests {
                     TestSemaphore::new(42).try_acquire().unwrap(),
                 ),
             };
-            let result = service.sample_envelope(&mut state);
-            assert_eq!(result.is_ok(), shouldkeep);
+
+            service.compute_sampling_decision(&mut state);
```

Comment:
nit: `compute_sampling_decision` is now a very thin wrapper around `should_keep_event()`, so we might as well only test `should_keep_event()` here. Alternatively, if we want to test that `sample_envelope()` correctly converts a sampling result to a `Result<...>`, revert to calling `sample_envelope()` here.

Reply:
Good observation. Instead of testing `should_keep_event` here, we can even remove this test, since there are unit tests for this function already. 

I'd probably test an entire `process` run then, that asserts we wire everything up correctly. Will look into this.